libraries/Microsoft.Bot.Connector.Streaming/Application/StreamingTransportClient.cs

```diff
@@ -0,0 +1,329 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO.Pipelines;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Connector.Streaming.Session;
+using Microsoft.Bot.Connector.Streaming.Transport;
+using Microsoft.Bot.Streaming;
+using Microsoft.Bot.Streaming.Transport;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace Microsoft.Bot.Connector.Streaming.Application
+{
+    /// <inheritdoc />
+    public abstract class StreamingTransportClient : IStreamingTransportClient
+    {
+        private readonly string _url;
+        private readonly RequestHandler _requestHandler;
+        private readonly TimeSpan _closeTimeout;
+        private readonly TimeSpan? _keepAlive;
+
+        private StreamingTransport _transport;
+        private TransportHandler _application;
+        private StreamingSession _session;
+
+        private CancellationTokenSource _disconnectCts;
+        private volatile bool _disposed = false;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="StreamingTransportClient"/> class.
+        /// </summary>
+        /// <param name="url">The server URL to connect to.</param>
+        /// <param name="requestHandler">Handler that will receive incoming requests to this client instance.</param>
+        /// <param name="closeTimeOut">Optional time out for closing the client connection.</param>
+        /// <param name="keepAlive">Optional spacing between keep alives for proactive disconnection detection. If null is provided, no keep alives will be sent.</param>
+        /// <param name="logger"><see cref="ILogger"/> for the client.</param>
+        protected StreamingTransportClient(string url, RequestHandler requestHandler, TimeSpan? closeTimeOut = null, TimeSpan? keepAlive = null, ILogger logger = null)
+        {
+            if (string.IsNullOrEmpty(url))
+            {
+                throw new ArgumentNullException(nameof(url));
+            }
+
+            _url = url;
+            _requestHandler = requestHandler ?? throw new ArgumentNullException(nameof(requestHandler));
+            _closeTimeout = closeTimeOut ?? TimeSpan.FromSeconds(15);
+            _keepAlive = keepAlive;
+            Logger = logger ?? NullLogger.Instance;
+        }
+
+        /// <inheritdoc />
+        public event DisconnectedEventHandler Disconnected;
```

Comment:
In the new code paths, are we using this event internally?

Reply:
i.e. if we have a bot that uses the new streaming library with named pipes, will this event be used? Wanted to confirm